WordPress/WordPressTest/ContextManagerTests.swift

```diff
@@ -5,74 +5,64 @@ import CoreData
 
 class ContextManagerTests: XCTestCase {
-    var contextManager: ContextManagerMock!
+    let storeURL = URL(fileURLWithPath: NSTemporaryDirectory()).appendingPathComponent("ContextManagerTests.sqlite")
 
-    override func setUp() {
-        super.setUp()
-
-        contextManager = ContextManagerMock()
+    override func setUpWithError() throws {
+        if FileManager.default.fileExistsAtURL(storeURL) {
+            try FileManager.default.removeItem(at: storeURL)
+        }
     }
 
-    func testIterativeMigration() {
-        let model19Name = "WordPress 19"
+    func testIterativeMigration() throws {
+        var objectID: NSManagedObjectID? = nil
 
-        // Instantiate a Model 19 Stack
-        startupCoredataStack(model19Name)
+        try prepareForMigration(withModelName: "WordPress 19") { context in
+            // Insert a Theme Entity
+            let objectOriginal = NSEntityDescription.insertNewObject(forEntityName: "Theme", into: context)
+            try context.obtainPermanentIDs(for: [objectOriginal])
+            try context.save()
 
-        let mocOriginal = contextManager.mainContext
-        let psc = contextManager.persistentStoreCoordinator
+            XCTAssertFalse(objectOriginal.objectID.isTemporaryID, "Should be a permanent object")
+            objectID = objectOriginal.objectID
 
-        // Insert a Theme Entity
-        let objectOriginal = NSEntityDescription.insertNewObject(forEntityName: "Theme", into: mocOriginal)
-        try! mocOriginal.obtainPermanentIDs(for: [objectOriginal])
-        try! mocOriginal.save()
-
-        let objectID = objectOriginal.objectID
-        XCTAssertFalse(objectID.isTemporaryID, "Should be a permanent object")
+            try XCTAssertThrowsError(
+                WPException.objcTry({
+                    objectOriginal.value(forKey: "author")
+                }),
+                "Theme.author doesn't exist in WordPress 19"
+            )
+        }
 
         // Migrate to the latest
-        let persistentStore = psc.persistentStores.first!
-        try! psc.remove(persistentStore)
-
-        let standardPSC = contextManager.standardPSC
-
-        XCTAssertNotNil(standardPSC, "New store should exist")
-        XCTAssertTrue(standardPSC.persistentStores.count == 1, "Should be one persistent store.")
-
-        // Verify if the Theme Entity is there
-        let mocSecond = NSManagedObjectContext(concurrencyType: NSManagedObjectContextConcurrencyType.mainQueueConcurrencyType)
-        mocSecond.persistentStoreCoordinator = standardPSC
-        let object = try! mocSecond.existingObject(with: objectID)
-
+        let contextManager = ContextManager(modelName: ContextManagerModelNameCurrent, store: storeURL)
+        let object = try contextManager.mainContext.existingObject(with: XCTUnwrap(objectID))
         XCTAssertNotNil(object, "Object should exist in new PSC")
+        XCTAssertNoThrow(object.value(forKey: "author"), "Theme.author exists in current version")
```

Comment:
What do you think of this change to the failure message, to make it clear that if `object.value...` throws, then it means that "author" doesn't exist and that goes against our expectation?
 
```suggestion
        XCTAssertNoThrow(object.value(forKey: "author"), "Theme.author should exist in current model version, but we were unable to fetch it")
```